L1Trigger/L1TTrackMatch/plugins/L1TkMuonProducer.cc

```diff
@@ -0,0 +1,712 @@
+    TFile* fIn_theta = TFile::Open(fIn_theta_name.c_str());
+    TFile* fIn_phi = TFile::Open(fIn_phi_name.c_str());
+    dwcorr_ = std::unique_ptr<L1TkMuCorrDynamicWindows>(new L1TkMuCorrDynamicWindows(bounds, fIn_theta, fIn_phi));
+
+    // files can be closed since the correlator code clones the TF1s
+    fIn_theta->Close();
+    fIn_phi->Close();
+
+    // FIXME: more initialisation using the parameters passed from the cfg
+    dwcorr_->set_safety_factor(iConfig.getParameter<double>("final_window_factor"));
+    dwcorr_->set_sf_initialrelax(iConfig.getParameter<double>("initial_window_factor"));
+
+    dwcorr_->set_relaxation_pattern(iConfig.getParameter<double>("pt_start_relax"),
+                                    iConfig.getParameter<double>("pt_end_relax"));
+
+    dwcorr_->set_do_relax_factor(iConfig.getParameter<bool>("do_relax_factors"));
+
+    //
+    dwcorr_->set_n_trk_par(iConfig.getParameter<int>("n_trk_par"));
+    dwcorr_->set_min_trk_p(iConfig.getParameter<double>("min_trk_p"));
+    dwcorr_->set_max_trk_aeta(iConfig.getParameter<double>("max_trk_aeta"));
+    dwcorr_->set_max_trk_chi2(iConfig.getParameter<double>("max_trk_chi2"));
+    dwcorr_->set_min_trk_nstubs(iConfig.getParameter<int>("min_trk_nstubs"));
+    dwcorr_->set_do_trk_qual_presel(true);
+  }
+
+  if (bmtfMatchAlgoVersion_ == kMantra) {
+    std::string fIn_bounds_name = iConfig.getParameter<edm::FileInPath>("mantra_bmtfcorr_boundaries").fullPath();
+    std::string fIn_theta_name = iConfig.getParameter<edm::FileInPath>("mantra_bmtfcorr_theta_windows").fullPath();
+    std::string fIn_phi_name = iConfig.getParameter<edm::FileInPath>("mantra_bmtfcorr_phi_windows").fullPath();
+
+    auto bounds = L1TkMuMantra::prepare_corr_bounds(fIn_bounds_name, "h_dphi_l");
+
+    TFile* fIn_theta = TFile::Open(fIn_theta_name.c_str());
+    TFile* fIn_phi = TFile::Open(fIn_phi_name.c_str());
+
+    mantracorr_barr_ = std::unique_ptr<L1TkMuMantra>(new L1TkMuMantra(bounds, fIn_theta, fIn_phi, "mantra_barrel"));
+
+    fIn_theta->Close();
+    fIn_phi->Close();
+
+    // settings : NB : now hardcoded, could be read from cfg
+    mantracorr_barr_->set_safety_factor(0.5, 0.5);
+    mantracorr_barr_->setArbitrationType("MaxPt");
+  }
+
+  if (omtfMatchAlgoVersion_ == kMantra) {
+    std::string fIn_bounds_name = iConfig.getParameter<edm::FileInPath>("mantra_omtfcorr_boundaries").fullPath();
+    std::string fIn_theta_name = iConfig.getParameter<edm::FileInPath>("mantra_omtfcorr_theta_windows").fullPath();
+    std::string fIn_phi_name = iConfig.getParameter<edm::FileInPath>("mantra_omtfcorr_phi_windows").fullPath();
+
+    auto bounds = L1TkMuMantra::prepare_corr_bounds(fIn_bounds_name, "h_dphi_l");
+
+    TFile* fIn_theta = TFile::Open(fIn_theta_name.c_str());
+    TFile* fIn_phi = TFile::Open(fIn_phi_name.c_str());
+
+    mantracorr_ovrl_ = std::unique_ptr<L1TkMuMantra>(new L1TkMuMantra(bounds, fIn_theta, fIn_phi, "mantra_overlap"));
```

Comment:
use make_unique (if this needs to be a pointer at all)

Reply:
done